ouroboros-network/src/Ouroboros/Network/Subscription/Dns.hs

```diff
@@ -0,0 +1,265 @@
+{-# LANGUAGE CPP                 #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+
+
+module Ouroboros.Network.Subscription.Dns
```

Comment:
Ah, perhaps it's just this module that is the RFC impl and not the other part? Is that right, it's fully encapsulated here, or is the policy also partly how the `SubscriptionTarget` is consumed?

Reply:
The creation of new connections in `Common.hs` and timeouts is very much a product of rfc8305 so it is not limited to `Dns.hs`.